A script to generate user module files

ivagljiva opened the pull request:
This PR introduces a simple little script that can automatically format user-defined module files. 

Most of the module information has to be passed to the script as command-line parameters, but the enzymes (and their ORTHOLOGY and ANNOTATION_SOURCE) values are passed in using a new artifact, the enzymes-txt, which is simply a tab-delimited file that looks like this:
```
enzyme	source	orthology
K01657	KOfam
K01658	KOfam
PF06603.14	METABOLISM_HMM	UpxZ
COG1362	COG20_FUNCTION	Aspartyl aminopeptidase
TIGR01709.2	TIGRFAM	type II secretion system protein GspL
```
(Note that KOfam ORTHOLOGY sources are allowed to be blank, and the script will fill these in by looking up the KO definition in the `ko_list.txt` file in the KEGG data directory.)

Here is an example of how I can run the script (using the above enzymes-txt file) to generate the fake user-defined pathway described in [the user modules help doc](https://merenlab.org/software/anvio/help/main/artifacts/user-modules-data/):
```
anvi-script-gen-user-module-file -I "UD0023" \
                  -e enzymes.txt \
                  -n "Frankenstein pathway for demo purposes" \
                  -c "User modules; Demo set; Frankenstein metabolism" \
                  -d "K01657+K01658 PF06603.14,(COG1362 TIGR01709.2)"
```

That produces a file called `UD0023` which looks like this:
```
ENTRY       UD0023
NAME        Frankenstein pathway for demo purposes
DEFINITION  K01657+K01658 PF06603.14,(COG1362 TIGR01709.2)
ORTHOLOGY   K01657  anthranilate synthase component I [EC:4.1.3.27]
            K01658  anthranilate synthase component II [EC:4.1.3.27]
            PF06603.14  UpxZ
            COG1362  Aspartyl aminopeptidase
            TIGR01709.2  type II secretion system protein GspL
CLASS       User modules; Demo set; Frankenstein metabolism
ORTHOLOGY   K01657  KOfam
            K01658  KOfam
            PF06603.14  METABOLISM_HMM
            COG1362  COG20_FUNCTION
            TIGR01709.2  TIGRFAM
\\\
```

If you do not provide a module definition with `-d`, then a definition will automatically be generated with each enzyme in the input file as a different step. For example:
```
anvi-script-gen-user-module-file -I "UD0023" \
                  -e input_txt_files/test_enzymes.txt \
                  -n "Frankenstein pathway for demo purposes" \
                  -c "User modules; Demo set; Frankenstein metabolism"
```
produces
```
ENTRY       UD0023
NAME        Frankenstein pathway for demo purposes
DEFINITION  K01657 K01658 PF06603.14 COG1362 TIGR01709.2
ORTHOLOGY   K01657  anthranilate synthase component I [EC:4.1.3.27]
            K01658  anthranilate synthase component II [EC:4.1.3.27]
            PF06603.14  UpxZ
            COG1362  Aspartyl aminopeptidase
            TIGR01709.2  type II secretion system protein GspL
CLASS       User modules; Demo set; Frankenstein metabolism
ORTHOLOGY   K01657  KOfam
            K01658  KOfam
            PF06603.14  METABOLISM_HMM
            COG1362  COG20_FUNCTION
            TIGR01709.2  TIGRFAM
\\\
```
This could be used for creating "signature" modules, in which each enzyme is not technically part of a metabolic pathway, but instead the module represents a functionally-related set of enzymes (like all tRNA modification enzymes, for instance).

@semiller10 and @meren , what do you think of this script? Is it broadly useful enough to be pushed to the main branch? @semiller10, does it work well with the strategy you are using to create your user-defined modules?

Happy to hear any other suggestions that either of you may have :)

If you think it's worth it, I'll add the relevant documentation to the codebase before merging.

Thanks for taking a look!

semiller10 reviewed (approved):
This certainly streamlines the creation of module files as much as possible. I like the depth of the artifact pages you've created -- very thorough explanations of the terminology and file/directory structures. I'm not proofreading the code :)

ivagljiva commented:
Given that it will be a while before I implement the changes discussed in #1873, we decided to merge this script to master to give people an easier way to generate their module files (in the meantime).

I have updated the artifact name to be `enzymes-list-for-module` so that it doesn't conflict with the existing `enzymes-txt`, and added the relevant documentation.
